src/sysusers/sysusers.c

```diff
@@ -758,9 +762,11 @@ static int write_files(void) {
         if (r < 0)
                 goto finish;
 
+#ifdef HAVE_GSHADOW
         r = write_temporary_gshadow(gshadow_path, &gshadow, &gshadow_tmp);
         if (r < 0)
                 goto finish;
+#endif
```

Comment:
not that it matters much, but I think it's a good idea to place the `#ifdef`fery always in the "inner" function they mask out, turning it into a NOP, so that the ifdeffery leaks into as little code as possible, if you follow what I mean. Yes, write_temporary_gshadow() only has a single caller, but stilll...